### ISSUE TITLE
app/Main.hs:77:5: error:  Couldn't match type ‘[Char]’ with ‘Text’

### ISSUE THREAD

**Issue by mrueg** [hadolint/hadolint#213]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
hadolint-1.7.0 builds with ghc-8.0.2
### Actual behavior
```
 * hadolint-1.7.0.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                          [ ok ]
>>> Unpacking source...
>>> Unpacking hadolint-1.7.0.tar.gz to /var/tmp/portage/dev-util/hadolint-1.7.0/work
>>> Source unpacked in /var/tmp/portage/dev-util/hadolint-1.7.0/work
>>> Preparing source in /var/tmp/portage/dev-util/hadolint-1.7.0/work/hadolint-1.7.0 ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-util/hadolint-1.7.0/work/hadolint-1.7.0 ...
 * Using cabal-1.24.2.0.
 * Prepending /usr/lib64/ghc-8.0.2 to LD_LIBRARY_PATH
/usr/bin/ghc -package Cabal-1.24.2.0 --make /var/tmp/portage/dev-util/hadolint-1.7.0/work/hadolint-1.7.0/Setup.hs -j5 +RTS -A256M -qb0 -RTS -threaded -dynamic -o setup
[1 of 1] Compiling Main             ( /var/tmp/portage/dev-util/hadolint-1.7.0/work/hadolint-1.7.0/Setup.hs, /var/tmp/portage/dev-util/hadolint-1.7.0/work/hadolint-1.7.0/Setup.o )
Linking setup ...
./setup configure --ghc --prefix=/usr --with-compiler=/usr/bin/ghc --with-hc-pkg=/usr/bin/ghc-pkg --prefix=/usr --libdir=/usr/lib64 --libsubdir=hadolint-1.7.0/ghc-8.0.2 --datadir=/usr/share/ --datasubdir=hadolint-1.7.0/ghc-8.0.2 --enable-tests --ghc-options=-j5 +RTS -A256M -qb0 -RTS --with-ar=x86_64-pc-linux-gnu-ar --ghc-option=-optc-march=native --ghc-option=-optc-O2 --ghc-option=-optc-pipe --ghc-option=-optl-Wl,-O1 --ghc-option=-optl-Wl,--as-needed --disable-executable-stripping --docdir=/usr/share/doc/hadolint-1.7.0 --verbose --enable-shared --enable-executable-dynamic --sysconfdir=/etc --disable-library-stripping
Configuring hadolint-1.7.0...
Dependency HUnit >=1.2: using HUnit-1.2.5.2
Dependency ShellCheck >=0.5.0: using ShellCheck-0.5.0
Dependency aeson -any: using aeson-1.0.2.1
Dependency base >=4.8 && <5: using base-4.9.1.0
Dependency bytestring >=0.10: using bytestring-0.10.8.1
Dependency containers -any: using containers-0.5.7.1
Dependency directory >=1.3.0: using directory-1.3.0.0
Dependency filepath -any: using filepath-1.4.1.1
Dependency gitrev >=1.3.1: using gitrev-1.3.1
Dependency hadolint -any: using hadolint-1.7.0
Dependency hspec -any: using hspec-2.2.4
Dependency language-docker >=6.0.1 && <7: using language-docker-6.0.1
Dependency megaparsec >=6.4: using megaparsec-6.5.0
Dependency optparse-applicative -any: using optparse-applicative-0.13.0.0
Dependency split >=0.2: using split-0.2.2
Dependency text -any: using text-1.2.2.1
Dependency void -any: using void-0.7.1
Dependency yaml -any: using yaml-0.8.15.2
Using Cabal-1.24.2.0 compiled by ghc-8.0
Using compiler: ghc-8.0.2
Using install prefix: /usr
Binaries installed in: /usr/bin
Libraries installed in: /usr/lib64/hadolint-1.7.0/ghc-8.0.2
Dynamic libraries installed in: /usr/lib64/x86_64-linux-ghc-8.0.2
Private binaries installed in: /usr/libexec
Data files installed in: /usr/share/hadolint-1.7.0/ghc-8.0.2
Documentation installed in: /usr/share/doc/hadolint-1.7.0
Configuration files installed in: /etc
No alex found
Using ar given by user at: /usr/bin/x86_64-pc-linux-gnu-ar
No c2hs found
No cpphs found
Using gcc version 6.4.0 found on system at: /usr/bin/x86_64-pc-linux-gnu-gcc
Using ghc version 8.0.2 given by user at: /usr/bin/ghc
Using ghc-pkg version 8.0.2 given by user at: /usr/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.17.3 found on system at: /usr/bin/haddock-ghc-8.0.2
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at: /usr/bin/hpc
Using hsc2hs version 0.68.1 found on system at: /usr/bin/hsc2hs
No hscolour found
No jhc found
Using ld found on system at: /usr/bin/x86_64-pc-linux-gnu-ld
No lhc found
No lhc-pkg found
Using pkg-config version 0.29.2 found on system at: /usr/bin/pkg-config
Using strip version 2.29 found on system at: /usr/bin/strip
Using tar found on system at: /bin/tar
No uhc found
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-util/hadolint-1.7.0/work/hadolint-1.7.0 ...
./setup build
Building hadolint-1.7.0...
Preprocessing library hadolint-1.7.0...
[1 of 8] Compiling Paths_hadolint   ( dist/build/autogen/Paths_hadolint.hs, dist/build/Paths_hadolint.o )
[2 of 8] Compiling Hadolint.Bash    ( src/Hadolint/Bash.hs, dist/build/Hadolint/Bash.o )
[3 of 8] Compiling Hadolint.Rules   ( src/Hadolint/Rules.hs, dist/build/Hadolint/Rules.o )
[4 of 8] Compiling Hadolint.Formatter.Format ( src/Hadolint/Formatter/Format.hs, dist/build/Hadolint/Formatter/Format.o )
[5 of 8] Compiling Hadolint.Formatter.Checkstyle ( src/Hadolint/Formatter/Checkstyle.hs, dist/build/Hadolint/Formatter/Checkstyle.o )
[6 of 8] Compiling Hadolint.Formatter.Codeclimate ( src/Hadolint/Formatter/Codeclimate.hs, dist/build/Hadolint/Formatter/Codeclimate.o )
[7 of 8] Compiling Hadolint.Formatter.Json ( src/Hadolint/Formatter/Json.hs, dist/build/Hadolint/Formatter/Json.o )
[8 of 8] Compiling Hadolint.Formatter.TTY ( src/Hadolint/Formatter/TTY.hs, dist/build/Hadolint/Formatter/TTY.o )
Preprocessing executable 'hadolint' for hadolint-1.7.0...
[1 of 2] Compiling Paths_hadolint   ( dist/build/autogen/Paths_hadolint.hs, dist/build/hadolint/hadolint-tmp/Paths_hadolint.dyn_o )
[2 of 2] Compiling Main             ( app/Main.hs, dist/build/hadolint/hadolint-tmp/Main.dyn_o )

app/Main.hs:77:5: error:
    • Couldn't match type ‘[Char]’ with ‘Text’
      Expected type: Parser [IgnoreRule]
        Actual type: Parser [String]
    • In the second argument of ‘(<*>)’, namely ‘ignoreList’
      In the first argument of ‘(<*>)’, namely
        ‘LintOptions <$> version <*> outputFormat <*> ignoreList’
      In the expression:
        LintOptions <$> version <*> outputFormat <*> ignoreList <*> files
```
### Steps to reproduce the behavior

**Output of `hadolint --version` or
expected: hadolint 1.7.0

**Additional environment details (OS, stack version, etc.)**
Gentoo Linux

**Comment by lorenzo** [comment]
Ill try to do my best as las time, but we don’t officially support 8.0.2

**Comment by trofi** [comment]
I think this failure is cause by the use of `optparse-applicateive-0.13`. The error
```
app/Main.hs:77:5: error:
    • Couldn't match type ‘[Char]’ with ‘Text’
      Expected type: Parser [IgnoreRule]
        Actual type: Parser [String]
```
means that `IgnoreRule` is of unexpected type. Namely:
```haskell
type IgnoreRule = Text

data LintOptions = LintOptions
    { ignoreRules :: [IgnoreRule], ... }

parseOptions :: Parser LintOptions
parseOptions =
    LintOptions <$> -- CLI options parser definition
    ignoreList <*> ...
where
    ignoreList =
        many
            (strOption
                 (long "ignore" <> help "Ignore rule. If present, config file is ignored" <>
                  metavar "RULECODE"))
```

Note: `strOption` is required to yield `Text` here. This feature was added ony to `optparse-applicative-0.14` in https://github.com/pcapriotti/optparse-applicative/commit/ddcd183dbc4b337dacca1c478be3e4156a6f5849#diff-4191720afbea61cacc0c18ca54118978

**Comment by lorenzo** [comment]
@trofi, so setting the requirement to 0.14 would resolve this issue?

**Comment by trofi** [comment]
Yes, I think so.

**Comment by lorenzo** [comment]
I pushed a fixed directly to master, will release a new version of hadolint tonight
